### PR TITLE
proxy d+ us

### PR DESCRIPTION
**Pull request by yeahme49**

### CODE REVIEW

File: resources/lib/dplay.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
 import xbmcaddon
 import xbmcgui
 
+from urllib.parse import urlparse, urlencode
```
Comment (Dis90) [inline resources/lib/dplay.py:17]: I'm pretty sure urllib.parse doesn't exists in Python 2 https://stackoverflow.com/questions/50638366/url-parse-python2-7-equivalent

File: resources/service/proxy.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+import socket
+import requests
+import threading
+import http
+from http import HTTPStatus
+from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler
+from io import BytesIO
+
+from urllib.parse import urlparse, parse_qs, urlencode
```
Comment (Dis90) [inline resources/service/proxy.py:9]: I'm pretty sure urllib.parse doesn't exists in Python 2 https://stackoverflow.com/questions/50638366/url-parse-python2-7-equivalent

**Comment by matthuisman** [comment]
i just seen you only need to change the framerate in the file?
Kodi detects the correct framerate once playback starts... you may get a black screen for a second while it changes...

do you really need to change it at all?

Your proxy is also way over-complicated.

just use Requests in the proxy to fetch the file (like the old code)
then no need for any of the gzip_producer stuff.
also get ride of protocol_version = "HTTP/1.1"
It'll cause headaches if you don't set content-length and keep-alive stuff.
If you remove that, it'll use nice and simple 1.0 and close the connection once the file is finished.

I'll see if I can get a simple sample up that is Python2 / Python3 compatible :)

**Comment by yeahme49** [comment]
The issue is the screen blanks once playback starts, it blanks again before where a commercial would be (i have ad-free d+) and once again after where a commercial would be. so depending on the length of the show and number of commercial breaks there should be, i get 7-12 screen blanks during a show and it's pretty annoying. after changing it to be 29.970 in the m3u8 file the problems went away.

**Comment by matthuisman** [comment]
here is an example for you:
https://github.com/matthuisman/proxy.plugin.example

**Comment by yeahme49** [comment]
updated using example, tested and it appears to work. thank you for the help @matthuisman

**Pull request merged**